apps/account-api/src/controllers/v1/handles-v1.controller.ts

```diff
@@ -36,21 +37,39 @@ export class HandlesControllerV1 {
   constructor(
     private handlesService: HandlesService,
     private enqueueService: EnqueueService,
+    private blockchainService: BlockchainRpcQueryService,
   ) {
     this.logger = new Logger(this.constructor.name);
   }
 
+  /**
+   * Validates the provided handle by checking its validity using the blockchain service.
+   * If the handle is not valid, a BadRequestException is thrown.
+   *
+   * @param baseHandle - The handle to be validated.
+   * @throws {BadRequestException} If the handle is invalid.
+   * @returns {Promise<void>} A promise that resolves if the handle is valid.
+   */
+  async validateHandle(baseHandle: string): Promise<void> {
+    const isValidHandle = await this.blockchainService.isValidHandle(baseHandle);
+    if (isValidHandle.isFalse) {
+      throw new BadRequestException('Invalid handle provided!');
+    }
+  }
+
   @Post()
   @HttpCode(HttpStatus.OK)
   @ApiOperation({ summary: 'Request to create a new handle for an account' })
   @ApiOkResponse({ description: 'Handle creation request enqueued', type: TransactionResponse })
+  @ApiBadRequestResponse({ description: 'Invalid handle provided or Provided signature is not valid for the payload!' })
```

Comment:
These are updates to the error handling and don't break the API, so I think it is safe to continue with v1.